src/placeos-rest-api/controllers/edges.cr

```diff
@@ -0,0 +1,233 @@
+require "hound-dog"
+require "placeos-core/client"
+require "promise"
+
+require "./application"
+require "./systems"
+
+module PlaceOS::Api
+  class Edges < Application
+    base "/api/engine/v2/edges/"
+
+    before_action :check_admin, except: [:index, :show, :edge]
+    before_action :check_support, only: [:index, :show]
+
+    before_action :current_edge, only: [:destroy, :drivers, :show, :update, :update_alt]
+
+    skip_action :authorize!, only: [:edge]
+    skip_action :set_user_id, only: [:edge]
+    skip_action :check_oauth_scope, only: [:edge]
+
+    getter current_edge : Model::Edge { find_edge }
+
+    class_getter connection_manager : ConnectionManager { ConnectionManager.new(core_discovery) }
+
+    # Validate the present of the id and check the secret before routing to core
+    ws("/control", :edge) do |socket|
+      token = params["token"]?
+
+      render status: :unprocessable_entity, json: {error: "missing 'token' param"} if token.nil? || token.presence.nil?
+
+      edge_id = Model::Edge.validate_token(token)
+      head status: :unauthorized if edge_id.nil?
```

Comment:
Maybe a 403 hear instead too as the token was passed, but is not valid.

Reply:
I think 401 is the go here, because the token is invalid